### PR TITLE
BNH-119 tenant filtering

### PR DESCRIPTION
**Pull request by TomHanmer29**
Created filters for the tenant list.
image
image
The user can select multiple filters which all stack on top of each other to refine the tenant list

### CODE REVIEW

**Comment by BenRamchandani** [comment]
Is there a way to check for e.g. everyone not on universal credit?

**Comment by TomHanmer29** [comment]
> Is there a way to check for e.g. everyone not on universal credit?

Not currently, I will add this though!

File: web/Controllers/AdminController.cs [hunk 1]
```diff
@@ -188,4 +187,19 @@ public ActionResult RejectAdminRequest(int userToAcceptId)
 
         return RedirectToAction("GetAdminList");
     }
+
+    [Authorize(Roles = "Admin")]
+    [HttpPost]
+    public IActionResult ReceiveTenantFilterList([FromForm] TenantListModel tenantListModel)
+    {
+        return RedirectToAction("GetFilteredTenants", tenantListModel);
+    }
```
Comment (BenRamchandani) [inline web/Controllers/AdminController.cs:196]: If you change the form method to GET can you get rid of this endpoint?

File: web/Services/AdminService.cs [hunk 2]
```diff
@@ -147,4 +149,28 @@ public UserDbModel GetUserFromId(int userId)
...
+                tenantQuery = currentFilter switch
+                {
+                    0 => tenantQuery.Where(t => t.Type == "Single"),
+                    1 => tenantQuery.Where(t => t.Type == "Couple"),
+                    2 => tenantQuery.Where(t => t.Type == "Family"),
+                    3 => tenantQuery.Where(t => t.HasPet == true),
+                    4 => tenantQuery.Where(t => t.ETT == true),
+                    5 => tenantQuery.Where(t => t.UniversalCredit == true),
+                    6 => tenantQuery.Where(t => t.HousingBenefits == true),
+                    7 => tenantQuery.Where(t => t.Over35 == true),
```
Comment (BenRamchandani) [inline web/Services/AdminService.cs:169]: Could turn this into an object with a nullable bool for each of these options. Slightly less succinct, but a bit simpler to use/understand

**Review by BenRamchandani — COMMENTED** [review]
> > Is there a way to check for e.g. everyone not on universal credit?
> 
> Not currently, I will add this though!

Sounds good. The code itself looks sensible, though you might have to change it to support yes/no/not filtered

**Comment by TomHanmer29** [comment]
image
image

**Review by BenRamchandani — APPROVED** [review]
This all looks good. You can get rid of the old tenant list endpoint if you want and only use the filtered one, just defaulting to an empty filter

**Comment by TomHanmer29** [comment]
I've made a new ticket to improve the styling of these so I can do that as part of a whole refactor then :)

**Pull request merged**